src/logic/function/basic.lean

```diff
@@ -79,6 +79,20 @@ def injective.decidable_eq [decidable_eq β] (I : injective f) : decidable_eq α
 lemma injective.of_comp {g : γ → α} (I : injective (f ∘ g)) : injective g :=
 λ x y h, I $ show f (g x) = f (g y), from congr_arg f h
 
+lemma injective.dite (p : α → Prop) [decidable_pred p]
```

Comment:
Should this use `{_ : decidable_pred p}`? cc @kmill

Reply:
@jcommelin I'm not familiar enough with that pattern to comment. I see `if_true` and `if_false` use `{...}` -- is the idea that this helps `simp`? or to create a new goal when the `decidable_pred` is missing?

For the place this lemma is going to be used, I know `[...]` works, but that's through introducing the needed `decidable` instance ahead of time with something like `haveI := classical.dec_eq ι`.

Reply:
@jcommelin This pattern is the one described in the ["implicit instance arguments" library note](https://leanprover-community.github.io/mathlib_docs/notes.html#implicit%20instance%20arguments), right? As far as I can tell, that's used to speed up elaboration. Does this code look likely to have issues? (I haven't tested / can't tell.)